src/game_config.cpp

```diff
@@ -91,18 +91,28 @@ bool
 const bool& debug = debug_impl;
 
 void set_debug(bool new_debug) {
+    int severity;
+    lg::get_log_domain_severity("deprecation", severity);
+    int delta_severity; // change in severity
+    switch(severity) {
+        case -1: // LG_DISABLED, hence don't change
```

Comment:
Wouldn't it be better to use the severity enumeration directly instead of the hard-coded values together with comments?

Reply:
yeah I 100% agree! The issue is the severity enumeration isn't used anywhere yet (according to my IDE). I was going to try to create a separate pull request that replaces int with the enum everywhere

Reply:
@Wedge009 I have just realised that an issue with my current solution is that if the severity goes above 3 (which it can't currently) then my code won't reset it properly. This wouldn't be an issue if everything was using the enum. I have [pushed](https://github.com/wesnoth/wesnoth/pull/8068/commits/b0fa883182272bb2c975ff5e4b0302433306bd4e) a more robust solution that is probably better. What do you think?

Reply:
Closing this pull request due to enum change (see pull request #8076). I'll make a new pull request (that is also more clear) for this issue that that uses the Enum once #8076 gets merged